tools/generator/templates/player/python/interface.cc.jinja2

```diff
@@ -0,0 +1,363 @@
+    if (PyErr_Occurred())
+        throw 42;
+    return out;
+}
+
+template <typename CxxType>
+std::vector<CxxType> python_to_cxx_array(PyObject* in)
+{
+    if (!PyList_Check(in))
+    {
+        PyErr_SetString(PyExc_TypeError, "a list is required");
+        throw 42;
+    }
+
+    std::vector<CxxType> out;
+    unsigned int size = PyList_Size(in);
+
+    for (unsigned int i = 0; i < size; ++i)
+        out.push_back(python_to_cxx<PyObject*, CxxType>(PyList_GET_ITEM(in, i)));
+
+    return out;
+}
+{% for enum in game.enum %}
+
+{{ enum.enum_summary|cxx_comment }}
+
+template<>
+PyObject* cxx_to_python<PyObject*, {{ enum.enum_name }}>({{ enum.enum_name }} in)
+{
+    PyObject* name = PyUnicode_FromString("{{ enum.enum_name }}");
+    PyObject* enm = PyObject_GetAttr(py_module, name);
+    if (enm == NULL)
+        throw 42;
+    PyObject* arglist = Py_BuildValue("(i)", (int) in);
+    PyObject* ret = PyObject_CallObject(enm, arglist);
+    Py_DECREF(name);
+    Py_DECREF(arglist);
+    Py_DECREF(enm);
+    return ret;
+}
+
+template <>
+{{ enum.enum_name }} python_to_cxx<PyObject*, {{ enum.enum_name }}>(PyObject* in)
+{
+  return ({{ enum.enum_name }}) python_to_cxx<PyObject*, int>(in);
+}
+{% endfor %}
+{% for struct in game.struct %}
+
+{{ struct.str_summary|cxx_comment }}
+
+template <>
+PyObject* cxx_to_python<PyObject*, {{ struct.str_name }}>({{ struct.str_name }} in)
+{
+    PyObject* tuple = PyTuple_New({{ struct.str_field|length }});
+    {% for field_name, field_type, field_comment in struct.str_field %}
+    PyTuple_SET_ITEM(tuple, {{ loop.index0 }}, ({{ field_type|cxx_to_python }}(in.{{ field_name }})));
+    {% endfor %}
+    {% if struct.str_tuple %}
+    return tuple;
+    {% else %}
+    PyObject* name = PyUnicode_FromString("{{ struct.str_name }}");
+    PyObject* cstr = PyObject_GetAttr(py_module, name);
+    Py_DECREF(name);
+    if (cstr == NULL)
+        throw 42;
+    PyObject* ret = PyObject_CallObject(cstr, tuple);
+    Py_DECREF(cstr);
+    Py_DECREF(tuple);
+    if (ret == NULL)
+        throw 42;
+    return ret;
+    {% endif %}
+}
+
+template <>
+{{ struct.str_name }} python_to_cxx<PyObject*, {{ struct.str_name }}>(PyObject* in)
+{
+    {{ struct.str_name }} out;
+    PyObject* i;
+    {% for field_name, field_type, field_comment in struct.str_field %}
+
+    {{ field_comment|cxx_comment }}
+        {% if struct.str_tuple %}
+    i = PyTuple_GetItem(in, {{ loop.index0 }});
+        {% else %}
+    i = cxx_to_python<PyObject*, int>({{ loop.index0 }});
+    i = PyObject_GetItem(in, i);
+        {% endif %}
+    if (i == NULL)
+        throw 42;
+    out.{{ field_name }} = {{ field_type|python_to_cxx }}(i);
+        {% if not struct.str_tuple %}
+    Py_DECREF(i);
+        {% endif %}
+    {% endfor %}
+
+    return out;
+}
+{% endfor %}
+
+{% for func in game.function %}
+
+{{ func.fct_summary|cxx_comment }}
+static PyObject* p_{{ func.fct_name }}(PyObject* self, PyObject* args)
```

Comment:
Please add a short comment what this does, it's inscrutable with all the templating.